tests/test_std.rs

```diff
@@ -622,6 +629,11 @@ fn combinations_with_replacement() {
         (0..3).combinations_with_replacement(0),
         <Vec<Vec<_>>>::new(),
     );
+    // Zero size on empty pool
+    it::assert_equal(
+        (0..0).combinations_with_replacement(0),
+        <Vec<Vec<_>>>::new(),
+    );
```

Comment:
This looks wrong. The empty vector is a valid combination of zero elements in `0..0`, so `(0..0).combinations_with_replacement(0)` should be `vec![vec![]]` (like `(0..0).combinations(0)` and `(0..0).permutations(0)`), not `<Vec<Vec<_>>>::new() == vec![]`.

Reply:
@andersk 

It seems that the original test cases of `combinations_with_replacement` has the same test cases, and I just made a copy of it.
Should fixing `combinations_with_replacement` and its test cases be included in this PR or should I make a separate PR?

Reply:
I found that python itertools implementation shows value 1 (single empty set) for following inputs.
```
>>> len(list(itertools.combinations_with_replacement([], 0)))
1
>>> len(list(itertools.combinations([], 0)))
1
>>>
```